tests/test_middleware_compatibility.py

```diff
@@ -0,0 +1,36 @@
+import asyncio
+
+from django.http import HttpResponse
+from django.test import AsyncRequestFactory, RequestFactory, TestCase
+
+from debug_toolbar.middleware import DebugToolbarMiddleware
+
+
+class MiddlewareSyncAsyncCompatibilityTestCase(TestCase):
+    def setUp(self):
+        self.factory = RequestFactory()
+        self.async_factory = AsyncRequestFactory()
+
+    def test_sync_mode(self):
+        request = self.factory.get("/")
+        middleware = DebugToolbarMiddleware(lambda x: HttpResponse(""))
+
+        # test middlware switches to sync (__call__) based on get_response type
+        self.assertFalse(asyncio.iscoroutinefunction(middleware))
+
+        response = middleware(request)
+        self.assertEqual(response.status_code, 200)
+
+    async def test_async_mode(self):
+        async def get_response(request):
+            return HttpResponse("<html><body>Django debug toolbar</body></html>")
+
+        middleware = DebugToolbarMiddleware(get_response)
+        request = self.async_factory.get("/")
+
+        # test middlware switches to async (__acall__) based on get_response type
```

Comment:
This comment should be in the docstring for the method definition rather than inline a few lines into the method.